tests/integration/test_alerts.py

```diff
@@ -2,6 +2,8 @@
 from datetime import timedelta
 
 import pytest
+from tests.integration import run_command
```

Comment:
`run_command` is no longer defined, it looks like it got move into the `command_runner` fixture?

Reply:
As looked closer at this, I noticed that command_runner set up and tore down the temp profile on each test, so did some experimenting with handling it a bit differently, and came up with this fixture that can be used in integration tests that both handles the temp profile problem (without having to swap out the default), as well as makes run_command/pyexpect unnecessary: 


```
@pytest.fixture(scope="session")
def integration_test_profile():
    """Creates a temporary profile to use for executing integration tests."""
    host = os.environ.get("C42_HOST") or "http://127.0.0.1:4200"
    username = os.environ.get("C42_USER") or "test_username@example.com"
    password = os.environ.get("C42_PW") or "test_password"
    try:
        create_profile(TEST_PROFILE_NAME, host, username, True)
    except:
        pass
    state = CLIState()
    state.profile = get_profile(TEST_PROFILE_NAME)
    state._sdk = py42.sdk.from_local_account(host, username, password)
    yield state
    delete_profile(TEST_PROFILE_NAME)
```

Then as an example, here's how we'd write a `send-to` integration test with your DataServer class:

```
def test_auditlogs_send_to(runner, integration_test_profile):
    with DataServer(protocol="TCP"):
        result = runner.invoke(
            cli,
            [
                "audit-logs",
                "send-to",
                "localhost:5140",
                "-b",
                begin_date_str,
                "-p",
                "TCP",
            ],
            obj=integration_test_profile,
        )
    assert result.exit_code == 0
```

Thoughts?

Reply:
This is nice.  

Though I have made some changes in the above code.

- We shouldn't be calling `py42` explicitly, let the application handle that.

- `try catch` with `pass` wasn't useful.

- Fetched current profile and set it back in tear down, to leave the system back in its original state, same as Jules had done in  `use_temp_file`.

- Set password to profile explicitly so that it doesn't prompt for it.

Reply:
The only drawback using `runner` is while specifying commands we can't specify the root command `code42`. We do this in unit tests, so as long as we are fine with that this is good.

Reply:
The try block for create_profile was for if the test profile already exists from a previous run that didn't get cleaned up (i.e. you ctrl-c out of a pytest run that you see everything failing).

Reply:
If we're using `set_password()` in an automated environment, we'll either need to make sure the keyring backend works, or set `.assume_yes=True` on the CLIState object before calling `set_password()` to it can fail back to the flat file. That's why I ended up just creating and setting the sdk manually on the CLIState object initially.

Reply:
Yea but without that it prompts for password, so then we are back to square one :)

Reply:
We _could_ just use the runner here to create the profile w/ password...

```
cmd = f"profile create -n {TEST_PROFILE_NAME} -u {username} -s {host} --password {password}"
runner.invoke(cli, cmd.split())
```

And if we wanted to be a bit safer about cleanup, instead of changing the default profile, we could just append f"--profile {TEST_PROFILE_NAME}" to each integration test cmd.

Reply:
Actually, I think I like the idea of just using the runner to create the profile with password in one-shot, since if we ever refactor any of the profile code, we shouldn't have to touch the tests (unless we make a breaking change to the user interaction too).

Reply:
Ugh, nevermind. It looks like I didn't add the `-y` option to `create` to bypass the prompt if keyring isn't available. So we'd need to update that first to make that work anyway.

Reply:
I think having integration tests around profile should be a task on its own and using profile in other features should be separate, so imo it does not matter how profile was/is created while testing other features.

Reply:
I just merged in adding -y to the `profile create` command, so the integration test profile fixture can be as simple as: 

```
@pytest.fixture(scope="session")
def integration_test_profile():
    """Creates a temporary profile to use for executing integration tests."""
    host = os.environ.get("C42_HOST") or "http://127.0.0.1:4200"
    username = os.environ.get("C42_USER") or "test_username@example.com"
    password = os.environ.get("C42_PW") or "test_password"
    runner.invoke(
        cli, 
        f"profile create -n {TEST_PROFILE_NAME} -u {username} -s {host} --password {password} -y".split()
    )
    yield
    runner.invoke(cli, f"profile delete {TEST_PROFILE_NAME} -y")
```

Then if we pass `--profile {TEST_PROFILE_NAME}` to each command we don't need to mess with switching the default.

Reply:
Instead of appending the profile name to every command what if we use `profile use` command to use the temporary profile?

Reply:
I'm just trying to avoid having the tests leave our profiles in an unexpected state. If we change the default and the tests fail to complete and clean up the fixture, then we need to manually switch it back. It's not a huge concern, but it's happened to me when I was trying these out and if we can avoid it it'd be nice.